### PR TITLE
Fix error handling and add tests

### PR DESCRIPTION
**Pull request by benj56**

### CODE REVIEW

File: src/utils/axiosSetup.ts [hunk 1]
```diff
@@ -40,10 +39,11 @@ export const createResponseRejectedInterceptor = (interceptorParams: Interceptor
...
       try {
         res = await interceptorParams.state.refreshingToken;
-      } catch (error: unknown) {
+      } catch (e) {
+        const error = e as AxiosError;
         // Check if refresh token expired, call tokenRefreshFailed() and then re-throw error
-        const status = (error as { raw?: { response?: { status?: number } } }).raw?.response?.status;
-        if (status === 497) {
+        const status = error.code;
+        if ((status ?? '').startsWith('497')) {
```
Comment (whosshell) [inline src/utils/axiosSetup.ts:46]: if(stats?.startsWith('497') is not sufficient? Or what is status type?

**Review by whosshell — DISMISSED** [review]

File: src/utils/axiosSetup.ts (hunk 1 above)
Comment (tomi-timetac) [reply src/utils/axiosSetup.ts:46]: Why even startsWith?

Comment (benj56) [reply src/utils/axiosSetup.ts:46]: It's `string | undefined`, I'm using starsWith just to be sure: 
image

Comment (tomi-timetac) [reply src/utils/axiosSetup.ts:46]: I believe the status itself is always 497 (integer) and this is added by postman (thats why its unknown, or OK for 200). So i would keep it ===.

Taken from axios docs:
`{
  // `status` is the HTTP status code from the server response
  status: 200,

  // `statusText` is the HTTP status message from the server response
  statusText: 'OK',
}
`

**Review by tomi-timetac — APPROVED** [review]

**Review by whosshell — APPROVED** [review]

**Pull request merged**